crypto/src/signature.rs

```diff
@@ -263,10 +263,71 @@ impl<T: Encode> SignatureOf<T> {
     }
 }
 
+/// Wrapper around [`SignatureOf`] used to reimplement [`Eq`], [`Ord`]
+/// to compare signatures only by there [`PublicKey`].
+#[allow(clippy::derive_hash_xor_eq)]
+#[derive(Deref, DerefMut, Hash, Encode, Decode, Serialize, Deserialize)]
+#[serde(transparent)]
+#[repr(transparent)]
+pub struct SignatureWrapperOf<T>(
+    #[deref]
+    #[deref_mut]
+    #[serde(bound(deserialize = "T: "))]
+    SignatureOf<T>,
+);
+
+impl<T> SignatureWrapperOf<T> {
+    #[inline]
+    fn inner(self) -> SignatureOf<T> {
+        self.0
+    }
+}
+
+impl<T> PartialEq for SignatureWrapperOf<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.0.public_key.eq(&other.0.public_key)
+    }
+}
+impl<T> Eq for SignatureWrapperOf<T> {}
+
+impl<T> PartialOrd for SignatureWrapperOf<T> {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        self.0.public_key.partial_cmp(&other.0.public_key)
+    }
+}
+impl<T> Ord for SignatureWrapperOf<T> {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        self.0.public_key.cmp(&other.0.public_key)
+    }
+}
+
+impl<T> fmt::Debug for SignatureWrapperOf<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl<T> Clone for SignatureWrapperOf<T> {
+    fn clone(&self) -> Self {
+        Self(self.0.clone())
+    }
+}
```

Comment:
Derive?

Reply:
Derive would add `T: Clone` bound and we don't need this.

Reply:
Because it's a phantom. If `#[derive(` weren't a macro, it'd be able to see that.